src/utils/functions.ts

```diff
@@ -0,0 +1,39 @@
+import * as fs from "fs";
+import path from "path";
+import { window, workspace } from "vscode";
+
+export const getSourceName = (contractTitle: string) => {
+  if (workspace.workspaceFolders === undefined) {
+    window.showErrorMessage("please open solidity project to work");
```

Comment:
This condition is not specifically checking if a solidity projects is opened or not we can modify this as `open a project` or add proper check condition that checks if a `hardhat project` is open or not